app/src/utils/data/contributions.ts

```diff
@@ -368,3 +375,103 @@ export function grantDonationListener(
     off: () => grantRoundManager.value.off(filter, listener),
   };
 }
+
+/**
+ * @notice given a userAddress and the contributions, return only the contributions for that address
+ * @param userAddress
+ * @param contributions
+ */
+export function filterContributionsByUserAddress(userAddress: string, contributions: Contribution[]) {
+  if (contributions && userAddress) {
+    return contributions.filter((contribution: Contribution) => {
+      return contribution.address === userAddress;
+    });
+  }
+  return undefined;
```

Comment:
Arguably a nit, but semantically we should be returning `null` instead, i.e. the value was set to `null`, whereas `undefined` means the variable hasn't been set yet